service/controller/clusterapi/v29/key/common.go

```diff
@@ -10,100 +10,66 @@ func IsDeleted(getter DeletionTimestampGetter) bool {
 }
 
 func OperatorVersion(getter LabelsGetter) string {
 	return getter.GetLabels()[label.OperatorVersion]
 }
 
-func PrivateRouteTableName(idx int) string {
-	// Since CloudFormation cannot recognize resource renaming, use non-indexed
-	// resource name for first AZ.
-	if idx < 1 {
-		return "PrivateRouteTable"
-	}
-	return fmt.Sprintf("PrivateRouteTable%02d", idx)
+func PrivateRouteTableName(az string) string {
+	return fmt.Sprintf("PrivateRouteTable-%s", az)
 }
 
-func PrivateSubnetName(idx int) string {
-	// Since CloudFormation cannot recognize resource renaming, use non-indexed
-	// resource name for first AZ.
-	if idx < 1 {
-		return "PrivateSubnet"
-	}
-	return fmt.Sprintf("PrivateSubnet%02d", idx)
+func PrivateSubnetName(az string) string {
+	return fmt.Sprintf("PrivateSubnet-%s", az)
 }
 
-func PrivateSubnetRouteTableAssociationName(idx int) string {
-	// Since CloudFormation cannot recognize resource renaming, use non-indexed
-	// resource name for first AZ.
-	if idx < 1 {
-		return "PrivateSubnetRouteTableAssociation"
-	}
-	return fmt.Sprintf("PrivateSubnetRouteTableAssociation%02d", idx)
+func PrivateSubnetRouteTableAssociationName(az string) string {
+	return fmt.Sprintf("PrivateSubnetRouteTableAssociation-%s", az)
 }
 
-func PublicSubnetName(idx int) string {
-	// Since CloudFormation cannot recognize resource renaming, use non-indexed
-	// resource name for first AZ.
-	if idx < 1 {
-		return "PublicSubnet"
-	}
-	return fmt.Sprintf("PublicSubnet%02d", idx)
+func PublicSubnetName(az string) string {
+	return fmt.Sprintf("PublicSubnet-%s", az)
 }
 
-func PublicRouteTableName(idx int) string {
-	// Since CloudFormation cannot recognize resource renaming, use non-indexed
-	// resource name for first AZ.
-	if idx < 1 {
-		return "PublicRouteTable"
-	}
-	return fmt.Sprintf("PublicRouteTable%02d", idx)
+func PublicRouteTableName(az string) string {
+	return fmt.Sprintf("PublicRouteTable-%s", az)
 }
 
-func PublicSubnetRouteTableAssociationName(idx int) string {
-	// Since CloudFormation cannot recognize resource renaming, use non-indexed
-	// resource name for first AZ.
-	if idx < 1 {
-		return "PublicSubnetRouteTableAssociation"
-	}
-	return fmt.Sprintf("PublicSubnetRouteTableAssociation%02d", idx)
+func PublicSubnetRouteTableAssociationName(az string) string {
+	return fmt.Sprintf("PublicSubnetRouteTableAssociation-%s", az)
 }
 
 func ReleaseVersion(getter LabelsGetter) string {
 	return getter.GetLabels()[label.ReleaseVersion]
 }
 
-func VPCPeeringRouteName(idx int) string {
-	// Since CloudFormation cannot recognize resource renaming, use non-indexed
-	// resource name for first AZ.
-	if idx < 1 {
-		return "VPCPeeringRoute"
+// SanitizeCFResourceName filters out all non-ascii alphanumberics from input
+// string.
+//
+// Example: SanitizeCFResourceName("abc-123") == "abc123"
+// Example2: SanitizeCFResourceName("Dear god why? щ（ﾟДﾟщ）") == "Deargodwhy"
+//
+func SanitizeCFResourceName(v string) string {
```

Comment:
Is this really necessary? Shouldn't we already have full control over the input?

Reply:
What do you mean?

Reply:
We sorted this while chatting.